### PR TITLE
Body snapshot

### PR DESCRIPTION
**Pull request by b00ris**

### CODE REVIEW

**Comment by AlexeyAkhunov** [comment]
Please change this PR to go to `master` not to main. Later will change it to `devel`

**Comment by AlexeyAkhunov** [comment]
Please rebase this PR to `devel` branch

**Comment by AlexeyAkhunov** [comment]
Can you also merge in GitHub action config, so that we can checks through this

**Comment by AskAlexSharov** [comment]
Hi. Here is 1 test - sometime failing on CI: 
```
--- FAIL: TestSnapshotMigratorStage (4.02s)
206
    snapshot_builder_test.go:169: 
207
        	Error Trace:	snapshot_builder_test.go:169
208
        	            				kv_util.go:55
209
        	            				snapshot_builder_test.go:168
210
        	Error:      	Not equal: 
211
        	            	expected: []byte{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xb, 0xb, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0}
212
        	            	actual  : []byte{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0}
213
        	            	
214
        	            	Diff:
215
        	            	--- Expected
216
        	            	+++ Actual
217
        	            	@@ -1,3 +1,3 @@
218
        	            	 ([]uint8) (len=40) {
219
        	            	- 00000000  00 00 00 00 00 00 00 0b  0b 00 00 00 00 00 00 00  |................|
220
        	            	+ 00000000  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
221
        	            	  00000010  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
222
        	Test:       	TestSnapshotMigratorStage
223
INFO [06-11|03:53:03.202] database closed (MDBX)                   mdbx=headers10
224
WARN [06-11|03:53:03.202] Hasn't stopped snapshot                  infohash=
225
INFO [06-11|03:53:03.202] Start seeding snapshot                   type=headers
226
INFO [06-11|03:53:04.188] database closed (MDBX)                   mdbx=chaindata
227
INFO [06-11|03:53:04.388] database closed (MDBX)                   mdbx=headers10
228
INFO [06-11|03:53:05.202] discovered 0 upnp devices 
229
INFO [06-11|03:53:07.308] Started seeding snapshot                 type=headers infohash=cfa8855ecdb58cf706bcae4145aee2196eb97d1d
230
INFO [06-11|03:53:07.308] Snapshot replaced 
231
INFO [06-11|03:53:07.308] SyncStages                               started=2
232
INFO [06-11|03:53:07.308] Prune db                                 current=0 new=10
233
INFO [06-11|03:53:07.308] Remove data                              from=0 to=10
234
INFO [06-11|03:53:07.309] Save CurrentHeadersSnapshotHash          new=cfa8855ecdb58cf706bcae4145aee2196eb97d1d new=10
235
INFO [06-11|03:53:07.312] CurrentHeadersSnapshotBlock commited     block=10
236
INFO [06-11|03:53:07.429] database closed (MDBX)                   mdbx=headers20
237
INFO [06-11|03:53:07.430] Stop seeding snapshot                    type=headers infohash=cfa8855ecdb58cf706bcae4145aee2196eb97d1d
238
INFO [06-11|03:53:07.466] Stopped seeding snapshot                 type=headers infohash=cfa8855ecdb58cf706bcae4145aee2196eb97d1d
239
INFO [06-11|03:53:07.466] Start seeding snapshot                   type=headers
240
INFO [06-11|03:53:11.595] Started seeding snapshot                 type=headers infohash=a47ceaaeb9dc76c47e5be56912a6da97fa309116
241
INFO [06-11|03:53:11.599] database closed (MDBX)                   mdbx=headers10
242
INFO [06-11|03:53:11.672] Snapshot replaced 
243
INFO [06-11|03:53:11.672] SyncStages                               started=2
244
INFO [06-11|03:53:11.672] Prune db                                 current=10 new=20
245
INFO [06-11|03:53:11.672] Remove data                              from=10 to=20
246
INFO [06-11|03:53:11.672] Save CurrentHeadersSnapshotHash          new=a47ceaaeb9dc76c47e5be56912a6da97fa309116 new=20
247
INFO [06-11|03:53:11.721] Removing old snapshot                    path=/tmp/TestSnapshotMigratorStageSyncMode395676729/001/snapshots/headers10
248
INFO [06-11|03:53:11.870] Removed old snapshot                     path=/tmp/TestSnapshotMigratorStageSyncMode395676729/001/snapshots/headers10 t=149.544723ms
249
INFO [06-11|03:53:11.870] CurrentHeadersSnapshotBlock commited     block=20
250
FAIL
251
FAIL	github.com/ledgerwatch/erigon/turbo/snapshotsync	13.903s
```

File: ethdb/kv/kv_snapshot.go [hunk 1]
```diff
@@ -347,20 +397,24 @@ func (s *snTX) CollectMetrics() {
...
-		return tx, nil
-	}
-	sn, ok := s.snapshots[bucket]
-	if !ok {
+	var tx ethdb.Tx
+	switch bucket {
+	case dbutils.HeadersBucket:
+		tx = s.headersTX
+	case dbutils.BlockBodyPrefix:
+		fallthrough
```
Comment (AlexeyAkhunov) [inline ethdb/kv/kv_snapshot.go:405]: The same thing could be achieved by specifying a comma separated list in the `case` I believe

**Comment by AlexeyAkhunov** [comment]
Please remove stray files in `ethdb/mdbx/dist`, and also remove changes in `tests/testdata`

**Pull request merged**